### PR TITLE
Migrate test_java_violations_reporter to native pytest

### PR DESCRIPTION
**Pull request by kasium**

### CODE REVIEW

File: tests/test_java_violations_reporter.py [hunk 1]
```diff
@@ -16,47 +19,45 @@
...
-        self.addCleanup(patch.stopall)
 
+class TestCheckstyleQualityReporterTest:
     def test_no_such_file(self):
         """Expect that we get no results."""
         quality = QualityReporter(checkstyle_driver)
 
         result = quality.violations("")
-        self.assertEqual(result, [])
+        assert not result
```
Comment (Bachmann1234) [inline tests/test_java_violations_reporter.py:47]: This is a weaker check. Lot of things can be falsey. 

Feels nitpicky, and I likely missed it in other prs. But I would prefer `assert result == []`

File: tests/test_java_violations_reporter.py [hunk 2]
```diff
@@ -16,47 +19,45 @@
...
 
     def test_no_java_file(self):
         """Expect that we get no results because no Python files."""
         quality = QualityReporter(checkstyle_driver)
         file_paths = ["file1.coffee", "subdir/file2.js"]
         for path in file_paths:
             result = quality.violations(path)
-            self.assertEqual(result, [])
+            assert not result
```
Comment (Bachmann1234) [inline tests/test_java_violations_reporter.py:55]: ditto

File: tests/test_java_violations_reporter.py [hunk 3]
```diff
@@ -79,55 +80,49 @@ def test_quality(self):
...
-        """
-        patch.stopall()
-
     def test_no_such_file(self):
         quality = QualityReporter(CheckstyleXmlDriver())
 
         # Expect that we get no results
         result = quality.violations("")
-        self.assertEqual(result, [])
+        assert not result
```
Comment (Bachmann1234) [inline tests/test_java_violations_reporter.py:113]: here as well

File: tests/test_java_violations_reporter.py [hunk 4]
```diff
@@ -79,55 +80,49 @@ def test_quality(self):
...
 
     def test_no_java_file(self):
         quality = QualityReporter(CheckstyleXmlDriver())
         file_paths = ["file1.coffee", "subdir/file2.js"]
         # Expect that we get no results because no Java files
         for path in file_paths:
             result = quality.violations(path)
-            self.assertEqual(result, [])
+            assert not result
```
Comment (Bachmann1234) [inline tests/test_java_violations_reporter.py:121]: and finally this

File: tests/test_java_violations_reporter.py [hunk 5]
```diff
@@ -276,41 +271,35 @@ def test_quality_pregenerated_report(self):
...
-        """
-        patch.stopall()
-
     def test_no_such_file(self):
         quality = QualityReporter(FindbugsXmlDriver())
 
         # Expect that we get no results
         result = quality.violations("")
-        self.assertEqual(result, [])
+        assert not result, []
```
Comment (Bachmann1234) [inline tests/test_java_violations_reporter.py:294]: whats going on with this one? I forget what assert does when you have the comma

File: tests/test_java_violations_reporter.py [hunk 6]
```diff
@@ -276,41 +271,35 @@ def test_quality_pregenerated_report(self):
...
 
     def test_no_java_file(self):
         quality = QualityReporter(FindbugsXmlDriver())
         file_paths = ["file1.coffee", "subdir/file2.js"]
         # Expect that we get no results because no Java files
         for path in file_paths:
             result = quality.violations(path)
-            self.assertEqual(result, [])
+            assert not result
```
Comment (Bachmann1234) [inline tests/test_java_violations_reporter.py:302]: another one

File: tests/test_java_violations_reporter.py [hunk 7]
```diff
@@ -417,41 +406,35 @@ def test_quality_pregenerated_report(self):
...
-        """
-        patch.stopall()
-
     def test_no_such_file(self):
         quality = QualityReporter(PmdXmlDriver())
 
         # Expect that we get no results
         result = quality.violations("")
-        self.assertEqual(result, [])
+        assert not result
```
Comment (Bachmann1234) [inline tests/test_java_violations_reporter.py:429]: one more

File: tests/test_java_violations_reporter.py [hunk 8]
```diff
@@ -417,41 +406,35 @@ def test_quality_pregenerated_report(self):
...
 
     def test_no_java_file(self):
         quality = QualityReporter(PmdXmlDriver())
         file_paths = ["file1.coffee", "subdir/file2.js"]
         # Expect that we get no results because no Java files
         for path in file_paths:
             result = quality.violations(path)
-            self.assertEqual(result, [])
+            assert not result
```
Comment (Bachmann1234) [inline tests/test_java_violations_reporter.py:437]: last one (Now that I got here I feel like I should have just said 'replace all of them please')

Comment (kasium) [reply tests/test_java_violations_reporter.py]: probably, but that made it easy for me :-D

File: tests/test_java_violations_reporter.py (hunk 7 above)
Comment (kasium) [reply tests/test_java_violations_reporter.py]: done

File: tests/test_java_violations_reporter.py (hunk 6 above)
Comment (kasium) [reply tests/test_java_violations_reporter.py]: done

File: tests/test_java_violations_reporter.py (hunk 5 above)
Comment (kasium) [reply tests/test_java_violations_reporter.py]: to be honest no clue, but that should be not like this. good catch

File: tests/test_java_violations_reporter.py (hunk 4 above)
Comment (kasium) [reply tests/test_java_violations_reporter.py]: done

File: tests/test_java_violations_reporter.py (hunk 3 above)
Comment (kasium) [reply tests/test_java_violations_reporter.py]: done

File: tests/test_java_violations_reporter.py (hunk 2 above)
Comment (kasium) [reply tests/test_java_violations_reporter.py:55]: done

File: tests/test_java_violations_reporter.py (hunk 1 above)
Comment (kasium) [reply tests/test_java_violations_reporter.py]: I agree. Let me change it

**Comment by Bachmann1234** [comment]
awesome, thanks!

**Pull request merged**